### DISCUSSION TITLE
Comment2GPT (Visual Studio extension)

### DISCUSSION THREAD

**Discussion by sapsari** [betalgo/openai#338] [Show and tell]
Thanks for sharing this library and supporting .netstandard2.0

I've created a Visual Studio extension using this, Comment2GPT. It basically auto completes your code while using your comment as your prompt.

https://marketplace.visualstudio.com/items?itemName=MerryYellow.Comment2GPT

![ezgif com-optimize (1)](https://github.com/betalgo/openai/assets/700243/0b99eae0-dfc7-4b3e-948c-c3ab0666e3b5)

**Comment by kayhantolga** [comment]
Congrats 🎉 , if it is open source I can add your repo to the Wiki page as a sample project.

**Reply by sapsari** [reply]
Unfortunately, it's not open source.

**Comment by damian-666** [comment]
I will try this ..looks good..
i cant use copilot or any MSFT ui that tryies to use AI even if they have so much almost there.. im using StartisBack and working with the indexing on that.



 VS has this code/feature search function and VoiceAssist.. if there ws a way to use that + voice + context to eliminate the need to click code , or feature, ( duh) and then click the UI. 
 i wan to say Turn on all Exceptions
 1.  use hotwords to make it listen. 
2.  use context to konw if its using the UI or coding ( obviously if you are in a code window)
       use Alpha, Bravo, Charlie Delat or whatever if there are multiple options that sound like you are choosing
       
3.  us the vocab and synonyms to limit the guesss if there is a hook for that..
4. let me command / query... 
therre are some open source realtime hot work detection out there.. that can interrupt commands.  

Voice Assist is realtime its just not domain specific.. if im coding it should use the completion metadata
its ui handling is horrible.  i cannot press a button with a tool tip.  

I can help design this.. i want to stand with a nice cardioid mic and at least debug... that  mic works will wiht voice assist..considering its not using context.

step, step, f5 etc..  by using that indexed command feature.   if i type step ,step in a debug window, via indexer it should find and exectute the commands while giving me visual feedback.

the feature is in the vs 2022preview.. with this is seems very possible.
JustSayit in julia has a kind of command scraper that does this, somewhat
Bard has voice but its not using its context and prediction to guess what im saying, its not mixed in the voice rec guesses.
The low latency text to speed might have only three words that are relevant in some of the context say Devstudio ( plus aliases)  =>File=> Alpha ( the first one, i dont love the 1, 2, 3 marks MSFT uses,  or File, Open, start saying the name of some file in the directory and its going to select that as a guess, then i say " Execute" or somehting.

ext.   it could be really fast , intuitive and i wont have to navigate , click 5 x chace menus,  etc.. remember custom keys that get reset., remember exact words,  .. Example in Start, If you say Launch Dev Studio, it doesnt work.  you have to say Open dev studio,  its doesnt learn, and they aren't fixing its ability to press buttons that have tooltips ( like most)  for 6 months.  the Copilot is a horrible slow mess that shows they want to waste your time and its much slower thatn Voice Access..  theres is  internal conflict over how to make money while killing our productivity.
usualy you are set to Bing.. shown ads.
